integration/spark/src/main/scala/org/apache/carbondata/indexserver/DistributedCountRDD.scala

```diff
@@ -69,15 +70,17 @@ class DistributedCountRDD(@transient ss: SparkSession, dataMapFormat: Distributa
       DataMapStoreManager.getInstance().clearInvalidSegments(dataMapFormat.getCarbonTable,
         dataMapFormat.getInvalidSegments)
     }
+    val globalQueue = SegmentProcessor.getInstance()
     val futures = if (inputSplits.length <= numOfThreads) {
       inputSplits.map {
-        split => generateFuture(Seq(split))
+        split => generateFuture(Seq(split), globalQueue)
       }
     } else {
       DistributedRDDUtils.groupSplits(inputSplits, numOfThreads).map {
-        splits => generateFuture(splits)
+        splits => generateFuture(splits, globalQueue)
       }
     }
+    globalQueue.emptyQueue()
```

Comment:
This will impact other jobs which might be still using globalQueue

Reply:
done